maas/resource_maas_subnet_ip_range_test.go

```diff
@@ -0,0 +1,186 @@
+package maas_test
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"terraform-provider-maas/maas"
+	"terraform-provider-maas/maas/testutils"
+	"testing"
+
+	"github.com/canonical/gomaasclient/entity"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccResourceMaasSubnetIPRange_basic(t *testing.T) {
+	// Setup ip range parameters
+	var ipRange entity.IPRange
+	subnet_name := acctest.RandomWithPrefix("test-subnet")
+	ipRangeAttrName := "maas_subnet_ip_range.test_ip_range"
+	range_type := "reserved"
+	comment := "test-comment"
+	ipStart := "10.88.88.1"
+	ipEnd := "10.88.88.50"
+	ipStartMod := "10.88.88.2"
+	ipEndMod := "10.88.88.49"
+	commentMod := "a-different-comment"
+
+	// Check functions
+	checks := []resource.TestCheckFunc{
+		testAccMAASSubnetIPRangeCheckExists(ipRangeAttrName, &ipRange),
+		resource.TestCheckResourceAttr(ipRangeAttrName, "type", range_type),
+		resource.TestCheckResourceAttr(ipRangeAttrName, "comment", comment),
+		resource.TestCheckResourceAttr(ipRangeAttrName, "start_ip", ipStart),
+		resource.TestCheckResourceAttr(ipRangeAttrName, "end_ip", ipEnd),
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testutils.PreCheck(t, nil) },
+		Providers:    testutils.TestAccProviders,
+		CheckDestroy: testAccCheckMAASSubnetIPRangeDestroy,
+		ErrorCheck:   func(err error) error { return err },
+		Steps: []resource.TestStep{
+			// Test creation
+			{
+				Config: testAccSubnetIPRangeExampleResource(
+					subnet_name,
+					range_type,
+					comment,
+					ipStart,
+					ipEnd,
+				),
+				Check: resource.ComposeTestCheckFunc(checks...),
+			},
+			// Test if resource drift is detected by modifying the IP range using the
+			// GO MAAS client, and then checking if the state is updated correctly on
+			// refresh
+			{
+				PreConfig: func() {
+					client := testutils.TestAccProvider.Meta().(*maas.ClientConfig).Client
+					params := entity.IPRangeParams{
+						Type:    range_type,
+						Comment: commentMod,
+						StartIP: ipStartMod,
+						EndIP:   ipEndMod,
+						Subnet:  strconv.Itoa(ipRange.Subnet.ID),
+					}
+					// Update the IP range to changed values that should be read into state
+					_, err := client.IPRange.Update(ipRange.ID, &params)
+					if err != nil {
+						t.Fatalf("Failed to update IP range: %s", err)
+					}
+				},
+				RefreshState:       true,
+				ExpectNonEmptyPlan: true,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(ipRangeAttrName, "start_ip", ipStartMod),
```

Comment:
since the test is modifying everything, let's also test all the other fields to confirm that they are also modified. Checking that subnet is the same is optional so feel free to check it or not.